### ISSUE TITLE
Загрузка из текстовых файлов (и сохранение) билетов loader.h / loader.cpp

### ISSUE THREAD

**Issue by stden** [levelp/CppTickets#9]
Формат хранения (текст):

```
имя_параметра=значение:имя_параметра_2=значение2;
```

Билеты разделены точкой с запятой **;** 
Имя параметра отделено от значения **=**
Параметры разделены двоеточиями **:**

**Comment by m1nsk** [comment]
Тут я буду описывать, что я вообще делаю, для тех кому мой код не будет понятен без объяснений и комментариев. Начну с того, что данные о билетах в текстовом файле нужно представлять в специальном формате однозначно понятном нашей программе, этот формат будет использоваться для внесения данных в базу и для чтения.
В текстовом файле данные о билетах будут разделяться между собой знаком ";".
Данные о параметрах знаком ":".
Каждый блок данных о параметрах будет состоять из символьного кода параметра, например: цена-"prc",место назначения "dst" и т.д. и значение параметра в символьном виде, например:"1.25","karaganda","submarine". и разделен знаком "=". Таким  образом весь блок параметра будет иметь вид "trnsp=airplane","dst=penza"
В результате мы получим строки вида:
dst=moskow:prc=115.5:trnsp=train:name=golubev;dst=penza:prc=55:trnsp=submarine:name=kondratiev; и т.д.
Поскольку параметры у разных транспортных средств могут отличаться, то для каждого класса придется делать свою реализацию загрузочника.

**Comment by alexey257** [comment]
Зашибись, это что-то наподобие csv, который понимает excel. Только я не помню, по-моему файл надо в ascii режиме открывать, в бинарном когда я сталкивался, мы еще какие-то другие файлы открывали, в которых пакеты протокола лежали друг за другом

**Comment by m1nsk** [comment]
Написал кусок кода для лоадера. Сначала думал, что в string.h есть то, что мне нужно, но оказалось не совсем то, так что в основном я изобретал string.h со своими свистелками попутно отстреливая себе ноги...обе ноги. Зато разобрался с векторами и классом string, хотя класс string тоже вектор=) мои функции сейчас разбивают текстовый файл определенного формата на данные от отдельных билетов, но они же мне помогут разбить строку данных билета на отдельные лексемы.
https://github.com/levelp/CppTickets/blob/master/loader.h хэдер
https://github.com/levelp/CppTickets/blob/master/loader.c соурс

**Comment by m1nsk** [comment]
Код чтения из файла готов, осталось только прописать инициализацию параметров, так что жду реализации классов видов транспорта.

**Comment by alexey257** [comment]
В проекте нету load.h и типа pPair. Примерно понял, как работает, спасибо за разбор как с вектором работать, потому что в книжках примеры с int, и мозг выносит, что итераторы и begin/end - это указатели, и если у нас тип уже указатель на строку, его тоже надо разыменовывать.
Наверное надо, чтобы данные из файла попадали в итоге в вектор в TicketOffice. Пользователь введет с клавы граничные значения для поиска, и ему на экран будет вывод списка билетов.

**Comment by m1nsk** [comment]
Да, прости, я все делал в одной файле, распихал в хедер-сорс только в гитхаб, даже не проверял
